pkg/controller/atlascustomrole/custom_role.go

```diff
@@ -11,55 +11,43 @@ import (
 	"github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api"
 
 	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/translation/customroles"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/translation/project"
 	akov2 "github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api/v1"
 	"github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/controller/workflow"
 )
 
 type roleController struct {
 	ctx       *workflow.Context
-	projectID string
+	project   *project.Project
 	service   customroles.CustomRoleService
 	role      *akov2.AtlasCustomRole
 	dpEnabled bool
 	k8sClient client.Client
 }
 
-func handleCustomRole(ctx *workflow.Context, k8sClient client.Client, akoCustomRole *akov2.AtlasCustomRole,
-	dpEnabled bool) workflow.Result {
+func handleCustomRole(ctx *workflow.Context, k8sClient client.Client, project *project.Project, service customroles.CustomRoleService, akoCustomRole *akov2.AtlasCustomRole, dpEnabled bool) workflow.Result {
 	ctx.Log.Debug("starting custom role processing")
 	defer ctx.Log.Debug("finished custom role processing")
 
 	r := roleController{
 		ctx:       ctx,
-		service:   customroles.NewCustomRoles(ctx.SdkClient.CustomDatabaseRolesApi),
+		project:   project,
+		service:   service,
 		role:      akoCustomRole,
 		dpEnabled: dpEnabled,
 		k8sClient: k8sClient,
 	}
 
-	result := r.Reconcile()
-	ctx.SetConditionFromResult(api.ReadyType, result)
-	return result
+	return r.Reconcile()
```

Comment:
Same here, removal `ctx.SetConditionFromResult(api.ReadyType, result)` makes the resource never be ready. I believe it also would feat better into the `idle()` transition instead of here. But, please, make sure move it to other place fix the problem.

Reply:
This refactor is not about any changes to the reconciliation logic. I will just fix the bug to leave the logic as it was with the new dual ref in place. We can make a separate PR if we need to fix the logic as well.

Reply:
updated